app/validators/mac_address_validator.rb

```diff
@@ -6,7 +6,11 @@ def validate_each(record, attribute, value)
   end
 
   def make_invalid(record, attribute)
-    record.errors.add(attribute, (options[:message] || _("is not a valid MAC address")))
+    err_msg = _("is not a valid MAC address")
+    # error message can already be present from the Net::Validations::normalize_mac method
+    if record.errors.full_messages.none? { |message| message.include?(err_msg) }
```

Comment:
Would it be sufficient to check if for the given attribute there's already any error message? I'd like to avoid doing string comparisons on error messages because it can easily break on translations. For example, when one is missing a translation while the other is translated. Or in a different way so it's not a substring.

Reply:
Thanks for the tip, it seems to do the trick.